src/components/Facets.tsx

```diff
@@ -29,17 +26,95 @@ export interface FacetsProps {
  * and {@link HierarchicalFacets} components can be used instead for more control over facet
  * configuration.
  *
+ * To override a single facet, use {@link StandardFacet}.
+ *
  * @param props - {@link FacetsProps}
  * @returns A React component for facets
  *
  * @public
  */
 export function Facets(props: FacetsProps) {
-  const { customCssClasses = {} } = props;
+  const { searchOnChange, children, customCssClasses = {} } = props;
+
+  const fieldIdToCustomFacetProps = new Map();
+  if (children) {
+    (Array.isArray(children) ? children : [children])
+      .forEach(child => fieldIdToCustomFacetProps.set(child.props.fieldId, child));
+  }
+
   return (
-    <div className={customCssClasses.facetsContainer}>
-      <StandardFacets/>
-      <NumericalFacets />
+    <div>
+      <FacetsProvider searchOnChange={searchOnChange} className={customCssClasses.facetsContainer}>
+        {facets => facets
+          .map((facet, i) => {
+            let facetType: FacetType = FacetType.STANDARD;
+            let facetProps: FacetProps = {
+              fieldId: facet.fieldId,
+              label: facet.displayName,
+            };
+            if (fieldIdToCustomFacetProps.has(facet.fieldId)) {
+              const customFacetElement: ReactElement = fieldIdToCustomFacetProps.get(facet.fieldId);
+              facetProps = customFacetElement.props;
+              facetType = getFacetTypeFromReactElementType(
+                (typeof customFacetElement.type === 'function') ? customFacetElement.type.name : '');
+            } else {
+              if (isStringFacet(facet)) {
+                facetType = FacetType.STANDARD;
+              } else if (isNumericalFacet(facet)) {
+                facetType = FacetType.NUMERICAL;
+              }
+            }
+
+            let facetComponent: ReactElement;
+            switch (facetType) {
+              case FacetType.NUMERICAL:
+                facetComponent = <div/>;
+                break;
+              case FacetType.STANDARD:
+                // fall through
+              default:
+                facetComponent = (
+                  <StandardFacetContent
+                    facet={facet}
+                    {...facetProps}
+                  />);
+            }
+
+            return (
+              <Fragment key={facet.fieldId}>
+                {facetComponent}
+                {(i < facets.length - 1) && <FilterDivider className={customCssClasses?.divider}/>}
+              </Fragment>
+            );
+          })
+        }
+      </FacetsProvider>
+      <NumericalFacets/>
     </div>
   );
-}
\ No newline at end of file
+}
+
+/**
+ * A component that displays a single standard facet. Use this to override the default rendering.
+ *
+ * @param props - {@link StandardFacetProps}
+ * @returns ReactElement
+ * @public
+ */
+export function StandardFacet(props: StandardFacetProps) { return null; }
```

Comment:
Oh I see this is just returning null though?

Reply:
I just use this to pass along the props from users. The actual rendering requires the facet so I created the StandardFacetContent to handle that

Reply:
Yeah I think we should just expose the facet as a prop, right? That would allow the component to be tested easily with arbitrary data (for example)? As well as eventually providing users with an easily modifiable base.

Reply:
The facet is currently handled by the FacetsProvider. I don't think exposing it as a prop on StandardFacet is what we want. Users won't be able to access the facets easily, unless they use the StandardFacet inside a FacetsProvider, which is what we are trying to blackbox for them.

Reply:
Ok